portal/package.json

```diff
@@ -3,6 +3,7 @@
 		"@mysten/bcs": "^1.0.2",
 		"@mysten/sui": "^1.1.2",
 		"base-x": "^4.0.0",
+		"parse-domain": "^8.0.2",
```

Comment:
You are right, that would break, because I was thinking of `walrus.site` as a domain of a portal `walrus` and with top level domain `site`. 🤔

To test that it fails I tested the `getDomain` with input `flatland.site`. The expected value would be `site`, but I received `flatland.site`. The same would be for `walrus.site` if it's listed. The domain will show up as `flatland.walrus.site`.

Reply:
In this case, would it make sense supporting domains like this though? i.e. `https://flatland.com` flatland is the suinsname, but what would the portal be?

Reply:
Moved this one to issue https://github.com/MystenLabs/walrus-sites/issues/124 after our call.